src/coreclr/jit/importercalls.cpp

```diff
@@ -8730,6 +8743,121 @@ void Compiler::impCheckCanInline(GenTreeCall*           call,
     }
 }
 
+//------------------------------------------------------------------------
+// impMinMaxIntrinsic: Imports a min or max intrinsic
+//
+// Arguments:
+//   method        - The handle of the method being imported
+//   callType      - The underlying type for the call
+//   intrinsicName - The intrinsic being imported
+//   tailCall      - true if the method is a tail call; otherwise false
+//   isMax         - true if the intrinsic computes the max; false for the min
+//   isMagnitude   - true if the intrinsic compares using the absolute value of the inputs
+//   isNumber      - true if the intrinsic propagates the number; false for NaN
+//
+GenTree* Compiler::impEstimateIntrinsic(CORINFO_METHOD_HANDLE method,
```

Comment:
CC. @dotnet/jit-contrib, this did end up being a runtime issue of sorts.

There's a general assumption made by R2R that the managed implementation (for Corelib in particular) will be consistent across all paths. That was strictly not the case for the `*Estimate` APIs where they are non-deterministic across hardware by design. -- Put another way, `R2R` essentially assumes that `Isa.IsSupported` is equivalent to `compOpportunisticallyDependsOn(Isa)` for corelib, but it assumes its equivalent to `compExactlyDependsOn(Isa)` for other libraries.

As such, the only fix we have today is to move it into the JIT where we can explicitly use `compExactlyDependsOn(Isa)` instead and force things to be annotated correctly. This has a minor added bonus in that it improves codegen around the method slightly for more complex scenarios and makes it more consistent with the handling of other math APIs.

Reply:
Can it be solved via ExactlyDependsOn attributes on C# side?

Reply:
Not from what I saw. The analyzer will actually warn in the case that you try to do something like that, because of the expectation that the paths are semantically equivalent.

It's detailed more in depth here with some explanations of what is safe vs not safe for corelib: https://github.com/dotnet/runtime/blob/40c024f69ab1c39982c043277e2bb90d2e510a24/docs/design/coreclr/botr/vectors-and-intrinsics.md?plain=1#L67

Doing it in the JIT, which is how all the other APIs that map down to 1 instruction are handled, ended up being the simpler approach.

Reply:
@tannergooding so does this PR solve the non-determinism? R2R still may prejit functions with these math functions, right?

Reply:
> so does this PR solve the non-determinism

No, the non-determinism is intentional for these APIs. They are explicitly `estimating` APIs that are allowed to differ in result based on the hardware they're running on.

> R2R still may prejit functions with these math functions, right?

The use of `compExactlyDependsOn` basically tells R2R that this method requires `Avx512F` and if it isn't present the functionality may differ. That will mark methods that call it as requiring themselves to be jitted.

If we made these `recursive` and therefore `mustExpand`, then R2R could leave a call to the method and only have the method jitted instead. But, that's a more complicated change since `ReciprocalSqrtEstimate` requires us to emit a call to `System.Math.Sqrt` in the worst case and requires adding handling for that to platforms that don't currently have hardware intrinsic support (like Arm32).

This is a relatively uncommon API (basically just used in specialized high-perf scenarios), so fixing the failing CI legs seemed more important than worrying about the handful of APIs that would be disqualified from R2R.